Gems/Atom/RHI/Vulkan/Code/Source/RHI/Device.cpp

```diff
@@ -691,14 +691,26 @@ namespace AZ
             }
             else
             {
+                // Temporarily initialize a CommandQueueContext in order to query the timestampValidBits of each queue family 
+                CommandQueueContext commandQueueContext;
+                CommandQueueContext::Descriptor commandQueueContextDescriptor;
+                commandQueueContextDescriptor.m_frameCountMax = RHI::Limits::Device::FrameCountMax;
+
+                RHI::ResultCode result = commandQueueContext.Init(*this, commandQueueContextDescriptor);
+
+                AZ_Assert(result == RHI::ResultCode::Success, "Failed to create command queues %s", physicalDevice.GetName().GetCStr());
+
                 for (uint32_t i = 0; i < RHI::HardwareQueueClassCount; ++i)
```

Comment:
This feels unnecessary. What happens if you move InitFeaturesAndLimits to be in Device::InitializeLimits after m_commandQueueContext is initialized. Or we could just move the code related to setting the m_queryTypesMask to InitializeLimits.

Reply:
@moudgils I've added a new commit with the Device::InitFeaturesAndLimits() method moved after m_commandQueueContext() is initialized. I agree, it's a much simpler change. Confirmed working on my device.